openwisp_controller/connection/api/views.py

```
@@ -66,6 +66,11 @@ class CommandListCreateView(BaseCommandView, ListCreateAPIView):
 
     def create(self, request, *args, **kwargs):
         self.assert_parent_exists()
+        if not DeviceConnection.objects.filter(
+            device_id=self.kwargs['device_pk'],
+            credentials__isnull=False,
+        ).exists():
+            raise NotFound(detail='Device has no credentials assigned.')
```

Can you raise the validation error from the serializer itself?